src/azul/drs.py

```diff
@@ -269,21 +266,30 @@ def get_object(self,
         depends on the access method specified.
         """
         if access_id is None:
-            return self._get_object(object_id, access_method=access_method)
+            return self._get_object(drs_uri, access_method)
         else:
-            return self._get_object_access(object_id, access_id, access_method=access_method)
-
-    def _get_object(self, object_id: str, access_method: AccessMethod) -> str:
-        url = self.url / object_id
+            return self._get_object_access(drs_uri, access_id, access_method)
+
+    def _get_url(self, drs_uri: str, access_id: Optional[str] = None) -> str:
+        # Note the desire to include any query params found in the DRS URI
+        # (eg 'drs://{netloc}/{id}?version=123') when building the URL.
+        parsed = furl(drs_uri)
+        assert parsed.scheme == 'drs', parsed.scheme
+        parsed.scheme = 'https'
+        file_uuid = str(parsed.path).lstrip('/')
```

Comment:
Nitpick: while "we all know" this is a file UUID, since we're in DRS land here I think it makes more sense to call this `drs_id` or `object_id` or something like that.